test-suite/src/test/require-jdk8/org/scalajs/testsuite/javalib/lang/MathTestOnJDK8.scala

```diff
@@ -191,12 +196,90 @@ object MathTestOnJDK8 extends JasmineTest with ExpectExceptions {
       expectThrows[ArithmeticException](Math.negateExact(Long.MinValue))
     }
 
-    it("should respond to `toIntExact") {
-      expect(Math.toIntExact(-2147483648L) == Int.MinValue)
-      expect(Math.toIntExact(2147483647L) == Int.MaxValue)
+    it("should respond to `toIntExact`") {
+      expect(Math.toIntExact(-2147483648L) == Int.MinValue).toBeTruthy
+      expect(Math.toIntExact(2147483647L) == Int.MaxValue).toBeTruthy
 
       expectThrows[ArithmeticException](Math.toIntExact(-2147483649L))
       expectThrows[ArithmeticException](Math.toIntExact(2147483648L))
     }
+
+    it("should respond to `floorDiv`") {
+      expect(Math.floorDiv(0, 1)).toEqual(0)
+      expect(Math.floorDiv(0, -1)).toEqual(0)
+      expect(Math.floorDiv(1, 1)).toEqual(1)
+      expect(Math.floorDiv(1, -1)).toEqual(-1)
+      expect(Math.floorDiv(1, 2)).toEqual(0)
+      expect(Math.floorDiv(1, -2)).toEqual(-1)
+      expect(Math.floorDiv(-1, 2)).toEqual(-1)
+      expect(Math.floorDiv(-1, -2)).toEqual(0)
+      expect(Math.floorDiv(1, Int.MaxValue)).toEqual(0)
+      expect(Math.floorDiv(1, Int.MinValue)).toEqual(-1)
+      expect(Math.floorDiv(-1, Int.MaxValue)).toEqual(-1)
+      expect(Math.floorDiv(-1, Int.MinValue)).toEqual(0)
+      expect(Math.floorDiv(Int.MaxValue, 1) == Int.MaxValue).toBeTruthy
+      expect(Math.floorDiv(Int.MaxValue, -1) == -Int.MaxValue).toBeTruthy
+      expect(Math.floorDiv(Int.MinValue, 1) == Int.MinValue).toBeTruthy
+      expect(Math.floorDiv(Int.MinValue, -1) == Int.MinValue).toBeTruthy
+
+      expect(Math.floorDiv(0L, 1L)).toEqual(0)
```

Comment:
Hum sorry, I'm only seeing this now:
for expectations on `Long`s, you should use the following "format":
```scala
expect(Math.floorDiv(0L, 1L) == 0L).toBeTruthy
```
`toEqual` is insufficiently precise on `Long`s.